plugins/BEdita/Core/src/Shell/Task/SetupConnectionTask.php

```diff
@@ -0,0 +1,315 @@
+
+use Cake\Console\Shell;
+use Cake\Database\Connection;
+use Cake\Database\Exception\MissingConnectionException;
+use Cake\Datasource\ConnectionManager;
+use Cake\Filesystem\File;
+use Cake\ORM\TableRegistry;
+use Cake\Utility\Hash;
+
+/**
+ * Task to setup connection.
+ *
+ * @since 4.0.0
+ */
+class SetupConnectionTask extends Shell
+{
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    public function getOptionParser()
+    {
+        $parser = parent::getOptionParser();
+        $parser
+            ->setDescription([
+                'Setup database connection.',
+            ])
+            ->addOption('config-file', [
+                'help' => 'Configuration file where updated connection config will be saved.',
+                'required' => false,
+                'default' => CONFIG . 'app.php',
+            ])
+            ->addOption('connection', [
+                'help' => 'Connection name to use.',
+                'short' => 'c',
+                'required' => false,
+                'default' => 'default',
+                'choices' => ConnectionManager::configured(),
+            ])
+            ->addOption('connection-driver', [
+                'help' => 'Driver to use for new connection. Useful for unattended runs.',
+                'required' => false,
+                'choices' => ['Mysql', 'Postgres', 'Sqlite'],
+            ])
+            ->addOption('connection-host', [
+                'help' => 'Database host for new connection. Useful for unattended runs.',
+                'required' => false,
+            ])
+            ->addOption('connection-port', [
+                'help' => 'Database port for new connection. Useful for unattended runs.',
+                'required' => false,
+            ])
+            ->addOption('connection-database', [
+                'help' => 'Database name (or path for SQLite) for new connection. Useful for unattended runs.',
+                'required' => false,
+            ])
+            ->addOption('connection-username', [
+                'help' => 'Database username for new connection. Useful for unattended runs.',
+                'required' => false,
+            ])
+            ->addOption('connection-password', [
+                'help' => 'Database password for new connection. Useful for unattended runs.',
+                'required' => false,
+            ])
+            ->addOption('connection-password-empty', [
+                'help' => 'Use empty password for new connection. Useful for unattended runs.',
+                'boolean' => true,
+            ]);
+
+        return $parser;
+    }
+
+    /**
+     * Configure connection.
+     *
+     * @return void
+     */
+    public function main()
+    {
+        $connectionName = $this->param('connection');
+        $connection = ConnectionManager::get($connectionName);
+        if (!($connection instanceof Connection)) {
+            $this->abort('Invalid connection object');
```

Comment:
[minor] is this check really necessary? 
we have it in many places, and is also never tested case...

Reply:
Mmh… Despite not strictly necessary right now, it might become useful in the future. `ConnectionManager::get()` is guaranteed to return a `ConnectionInterface` instance. Database connections are just one possible implementation of `ConnectionInterface`. For example, ElasticSearch connections will be different implementations of the same interface.

This is why I would leave it in a place where we are explicitly managing a database connection: it might cause issues in the future if a user runs `bin/cake bedita setup --connection elasticsearch`. I have added a test case to test this condition.